### PR TITLE
feat(docs): enabled tertiary nav in sidenav

### LINKED ISSUE #2906: Implement tertiary nav in the sidebar
Towards epic #2903 

Implement [tertiary nav seen here](https://github.com/patternfly/patternfly-org/issues/2883#issuecomment-1089225664) to list different components within each extension under the nav structure `Extensions`->`[extensionName]`->`[extensionComponentName]`.

FYI: Sidenav items are [created here](https://github.com/patternfly/patternfly-org/blob/main/packages/v4/patternfly-docs.config.js#L14) and [implemented here](https://github.com/patternfly/patternfly-org/blob/main/packages/v4/patternfly-docs.source.js#L12), nav items are mapped through & [nav generated here](https://github.com/patternfly/patternfly-org/blob/main/packages/theme-patternfly-org/components/sideNav/sideNav.js#L54-L94).

### PR DESCRIPTION
**Pull request by evwilkin**
Closes #2906 

This PR makes the following changes to the `documentation-framework`:
- enables tertiary nav items in the main sidenav
  - this is done by adding a new `subsection` frontmatter field that creates the expandable nav within an existing `section` to put the tertiary links into.

This PR also makes the following changes to the PF website:
_*Edit*: to be removed from this PR, keeping for now to give a visual of the changes in the preview link._
- moves existing `Contribute` pages to be tertiary pages within a new `Contribute` subsection inside of the `Get started` section.
- [removes the `Contribute` section from the PF sidenav](https://github.com/patternfly/patternfly-org/pull/3274/files#diff-a359fb1a65f39405fea8f620769f663a32ad3aab3a7914c9734565f8f73af061)

<details>
<summary>Technical details</summary>

_Suggested order of review:_

### [slugger.js](https://github.com/patternfly/patternfly-org/pull/3274/files#diff-3ee9ad54794ff5bf9b394665fa24365b2badc6bf215d4bf5c08245baf7baba31)
_helper function to build URL string_
- Updates the `makeSlug` helper function, which is used throughout the documentation-framework to build out a string for the page URLs (ex: `/get-started/about`)
  - This update allows the function to include the subsection for pages where it exists (ex: `/get-started/contribute/about`).

### [parseMd.js](https://github.com/patternfly/patternfly-org/pull/3274/files#diff-97bd431987a2e7dda0f66ad7f07ac6cf4eeebf21fbe44a1b45ef905968aa14ca)
_markdown metadata parsing_
- This file loops through the markdown files and reads the frontmatter metadata at the top of each file.
- These changes just allow it to look for & pick up the new `subsection` optional frontmatter, similar to what it already does for the existing `section` data.

### [routes.js](https://github.com/patternfly/patternfly-org/pull/3274/files#diff-ccdde6d29008148c60b9bca52586afeb5316cf5f1ac9351c766e66b2790533a8)
_organizes all pages/links into a JS object mirroring the final navigation_
- This file creates the js object `groupedRoutes` which contains the main sections of the sidenav and each docs page within these sections.
- The first half of these changes (lines 31-58) expands upon a single level of nesting in the nav: `accum[section]` (ex: `Get started => About`) to allow for an additional level of nesting within a section: `accum[section][subsection]` (ex: `Get started => Contribute => About`)
  - `section` === the main expandable section in the sidenav
  - `subsection` === an expandable section within a `section`, to create the tertiary nav
  - `id` === the page name
  - `sources` === tabs within a page (ex: React/React demos/HTML/HTML demos/Accessibility/Design guidelines)
- The second half of these changes (lines 105-125) maintains existing functionality (much of the code is simply rearranged) but loops through nested objects when it hits a subsection to avoid the existing `forEach` loops running into the dreaded `[object Object]`.

### [sideNav.js](https://github.com/patternfly/patternfly-org/pull/3274/files#diff-8e3c7674b90a7a160ac769d32b003a8c5ba29ee93599606701fd47312630fc0a)
_loops through JS object to build out website navigation_
- This file takes the `navItems` array [(aka sideNavItems - passed to the documentation-framework from the user)](https://github.com/patternfly/patternfly-org/blob/main/packages/v4/patternfly-docs/patternfly-docs.config.js#L18-L33) to create the main sidenav sections, then loops through the `groupedRoutes` object (created above in `routes.js`) to fill in all of the subsections/navitem links within each section.
- The main two changes in this file are:
  - pulling out the code returned from `SideNav` into a separate component `ExpandableNav` for readability
  - updating the final `.map` inside of `ExpandableNav` from simply returning a `NavItem` (when a sidenav section could only contain one layer of links) to a conditional that instead returns a nested ExpandableNav (if the current item is a subsection to be created) or a `NavItem` (as before, if the current item is not a subsection but just a normal link to be created).

</details>

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
@evwilkin I think this looks great! It works well as a POC, however not sure we should merge this until we've had a chance to make other IA changes. Would it make sense to just merge this to add the capability only and open a second PR to move pages around? Let's discuss.

File: packages/documentation-framework/components/sideNav/sideNav.js [hunk 1]
```diff
@@ -34,6 +39,42 @@ const NavItem = ({ text, href }) => {
...
+      isActive={isActive}
+      isExpanded={isActive}
+      className="ws-side-nav-group"
+      onClick={(event) => {
+        // Don't trigger for bubbled events from NavItems
+        if (!event.target.href) {
+          const isExpanded = event.currentTarget.classList.contains('pf-m-expanded');
+          // 1 === expand section, 0 === collapse section
+          trackEvent('sidenav_section_click', 'click_event', currentSection, isExpanded ? 0 : 1);
+        }
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/documentation-framework/components/sideNav/sideNav.js]: @wise-king-sullyman nice catch!  I was firing the event on each click of NavExpandable, and the `!event.target.href` check is preventing a click event on a navitem within an expandable section from bubbling up and triggering this event.

Now that these NavExpandable can be nested inside of each other for tertiary nav, I had to add a `stopPropagation` to keep a click on a subsection expandable section from also triggering a click event on the parent NavExpandable.

File: packages/documentation-framework/scripts/webpack/getHtmlWebpackPlugins.js [hunk 2]
```diff
@@ -22,7 +22,8 @@ async function getHtmlWebpackPlugin({
       // Don't prerender fullscreen pages (expensive!)
       prerendering: (isProd && !isFullscreen && !url.includes('extensions')) ? await prerender(url, pathPrefix) : null,
       // Don't use GA in dev mode
-      googleAnalyticsID: isProd ? googleAnalyticsID : false,
+      // googleAnalyticsID: isProd ? googleAnalyticsID : false,
+      googleAnalyticsID: googleAnalyticsID,
```
Comment (wise-king-sullyman) [inline packages/documentation-framework/scripts/webpack/getHtmlWebpackPlugins.js:26]: Just confirming that you are wanting to leave analytics on in dev, and also that you want to leave line 25 in but commented out.

**Review by wise-king-sullyman — COMMENTED** [review]
Looks good! One question:

**Review by wise-king-sullyman — DISMISSED** [review]
🚀

File: packages/documentation-framework/scripts/webpack/getHtmlWebpackPlugins.js (hunk 2 above)
Comment (evwilkin) [reply packages/documentation-framework/scripts/webpack/getHtmlWebpackPlugins.js]: No, thanks for calling this out

**Review by wise-king-sullyman — APPROVED** [review]
🔥

**Pull request merged**